### PR TITLE
Fixes for MII domain using domainInfo.ServerStartMode: secure

### PR DESCRIPTION
**Pull request by jshum2479**
This PR fixes if an MII domain is created by WDT with domainInfo.ServerStartMode: secure only.  

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13830/  (only SecureMode Test )
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13831   (entire parallel run )

### CODE REVIEW

File: operator/src/main/resources/scripts/introspectDomain.py [hunk 1]
```diff
@@ -2004,11 +2004,23 @@ def getRealListenPort(template):
...
+  secureModeEnabled = False
+
+  # Do not use domain.getSecurityConfiguration().getSecureMode()
+  # it will result in cie error in MII domain created by ServerStartMode: secure
+  # switched to use lsa() to avoid cie not exposing the function
+
+  cd('/SecurityConfiguration/' + domain.getName())
+  childs = ls(returnType='c', returnMap='true')
+  if 'SecureMode' in childs:
+    cd('SecureMode/NO_NAME_0')
```
Comment (lennyphan) [inline operator/src/main/resources/scripts/introspectDomain.py:2016]: Should this be NO_NAME_0 ?  Shouldn't it be the domain name?

Comment (jshum2479) [reply operator/src/main/resources/scripts/introspectDomain.py:2016]: This is offline, so it is NO_NAME_0

**Review by lennyphan — APPROVED** [review]

**Comment by jshum2479** [comment]
Reopen after fixing problem detected in integration test

**Review by lennyphan — APPROVED** [review]
Just didn't understand the last sentence of the comment, otherwise LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItProductionSecureMode.java [hunk 2]
```diff
@@ -152,11 +152,13 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
     createSecretWithUsernamePassword(encryptionSecretName, domainNamespace,
             "weblogicenc", "weblogicenc");
 
+    // Add ServerStartMode section in domainInfo section 
+    // instead of SecurityConfiguration in topology section
+    // This will trigger a new path while WebLogic configuration
```
[the remark this answers is not in the document]
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItProductionSecureMode.java:157]: While generating WebLogic configuration, the internal py script (model_wdt_mii_filter.py) goes thru different path when SecureMode is configured in DomainInfo section vs SecurityConfiguration section of the model file.

**Review by lennyphan — APPROVED** [review]

**Review by anpanigr — APPROVED** [review]

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**